### ISSUE TITLE
Username out of order in reply list

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#11255]
As reported by 🎶💖  Kira Macarthur 💖🎶 / @kiramac (0fb3db86-7009-46d6-9eea-485a8f7e6734):

_"I was posting in the tavern when I noticed this [see screenshot below]. There is no-one called @2648 who posted recently, but it's coming up first."_

I confirmed it - the @2648 Username appeared in the list when I typed "@" in the chat box.

There was no recent message from that player (not even one that had been hidden by mods). The player had posted about 9 hours earlier, so their message was well down the chat with many other people's messages ahead of them.

Possibly the fact that the Username is all digits might have caused it to be sorted to the top. That's just a guess at the moment. I haven't seen this for any other Username.

![Error](https://i.imgur.com/1mUorFt.png)

**Comment by citrusella** [comment]
I think I saw this once with a different numbered display name/username, but it must have only happened once and then I forgot to report. o.o Maybe that's a point in favor of all digits sorting to the top for some reason theory?

(I tried looking for what code could cause this and couldn't make heads or tails of whether I was on the right track. XP)

**Comment by rbstrachan** [comment]
@Alys @citrusella 

I've also had a look in a few places I thought might help, as well as ran searches for instances of "sort" and "tavern", etc. The only thing I found was [this .js file](https://github.com/HabitRPG/habitica/blob/20fc233ad7116385594deff19ae619059e21c2d8/website/server/libs/chat/group-chat.js#L14). I think it's for sorting messages, not the reply list, but I have no idea.

Do you have any idea where would be a good place to look? I've probably missed something obvious...

Thanks :)

**Comment by citrusella** [comment]
I tried looking at [autoComplete.vue](https://github.com/HabitRPG/habitica/blob/d406da4081353120afe71c9f9277e8c97c0116ab/website/client/components/chat/autoComplete.vue) and a few chat files (I think chat and tavern and maybe one other one but I was kind of looking in a flurry), but there was either no indication there either of why a numbered username (or whatever is the characteristic it's looking for) would overtake a more recent name or I just didn't understand the code.

**Comment by Alys** [comment]
It's weird. I can't replicate it by inserting a numerical Username into an old post in a group's chat. I'm not certain it is from having a numerical Username. I'm going to mark this as needs investigation. 

It's really low priority though because even if it happens again, it's at worst mildly annoying.

@rbstrachan You're right that the code you linked to is for sorting messages (it's called when the messages are fetched from the database to ensure they appear in chronological order).

@citrusella I can't find the code that might be responsible either, although I gave up looking at it pretty quickly. :)

Thanks for looking into this both of you. If you want to continue, you'd be welcome to but I'm not sure it's worth the time. :)

**Comment by rbstrachan** [comment]
I can't seem to find user **@2648** anymore, however, when I first looked in to this I noticed they were sending messages in Mandarin Chinese. 

Does that make a difference? I'm really lost on this one!

**Comment by Alys** [comment]
@rbstrachan I wouldn't expect the language to make a difference but I really don't know what's causing this bug. Feel free to drop this if you don't want to keep searching for the cause! :)

**Comment by Alys** [comment]
Another report of it from @geolojess:

"when I'm in the Tavern and I go to tag a user, the first two options are these odd strings of numbers that don't seem to belong to any users? Or at least users that haven't posted in the Tavern recently. The top two"

and @citrusella:

"Those numbers belong to users who have posted recently, just not in the sense that they should be ordered that way in the dropdown! ... Those two users were chatting about what countries they were from a few hours back, IIRC."

![image](https://user-images.githubusercontent.com/1495809/74775253-0f895880-52e1-11ea-8c6f-10237e312bac.png)

**Comment by veeeeeee** [comment]
As this is a relatively minor issue that's easily resolved by typing one additional letter to specify the intended recipient, we're closing to redirect contributor energy towards more impactful changes.

**Comment by citrusella** [comment]
So... I know this issue is closed and VERY low priority (hence why it was closed)... and it's possible I'm sniffing up the wrong tree (my JS is still very elementary), but I wanted to link something that felt suspect just because I was investigating for funsies.

https://github.com/HabitRPG/habitica/blob/d406da4081353120afe71c9f9277e8c97c0116ab/website/client/components/chat/autoComplete.vue#L158-L159

Assuming grabUserNames generates the list: I went and read the documentation for groupBy and it looks like it iterates through the collection in the order presented to it (...makes sense)--could that mean the user is for some reason at the top of the chat collection "handed" to this script? (Or am I foolish in thinking it happens that early?)

Again, I get this is low priority and closed and probably worth ignoring--I just saw that and had the feeling I was vaguely following its logic and felt like recording my thoughts in an old Github issue. DX
